src/input/virtual_input.h

```diff
@@ -59,14 +60,14 @@ class CVirtualInput : public InputInterface {
             else
                 data_tmpSize = static_cast<size_t>(recordBuffer->GetRingBufferReadAvailable());
 
-            uint8_t data_tmp[data_tmpSize];
+            uint8_t data_tmp [1024];
```

Comment:
What is the reason to use  `1024` instead of `data_tmpSize` here?

Reply:
Look at the `if` test above where `data_tmpSize <= 1024`.

Reply:
But what happens when the else branch is executed (line 61)?

Reply:
Then the `GetRingBufferReadAvailable()` is 1024 or less. No?
Why CPU cycles on yet another VLA call?

Reply:
I think we should anyway change this code to call `GetRingBufferReadAvailable` only once, but using a fixed size buffer.

Reply:
@mpbraendli I agree. Is this cleaner?
```c
        while (1) {
            size_t  data_tmpSize;
            uint8_t data_tmp [1024];
            size_t  available = (size_t) recordBuffer->GetRingBufferReadAvailable();

            if (available <= 0)
               break;
            data_tmpSize = std::min(sizeof(data_tmp), available);
            recordBuffer->getDataFromBuffer(data_tmp, data_tmpSize);
            rawStream.write((char *) data_tmp, data_tmpSize);
        }
```

BTW, why 2 copies; why not write the head of the ring-buffer directly?

Reply:
I like that approach better, thanks.

We can't access the ringbuffer data directly because it handles threadsafe accesses.

Reply:
Perfect!
@gvanem You can do it in this way.